README.md

```diff
@@ -1,121 +1,183 @@
-# Purpose
+# Table of Contents
 
-A custom bootloader for the Commander X16 ATtiny861 based System Management Controller (SMC).
+- [Project Description](#project-description)
+- [How the Bootloader Works](#how-the-bootloader-works)
+    - [Introduction](#introduction)
+    - [Bootloader Entry Points](#bootloader-entry-points)
+    - [Transmitting the new Firmware](#transmitting-the-new-firmware)
+    - [Verifying the new Firmware](#verifying-the-new-firmware)
+    - [Fail-Safe](#fail-safe)
+- [SMC Memory Map](#smc-memory-map)
+- [Building the Project](#building-the-project)
+- [Installing the Bootloader](#installing-the-bootloader)
+    - [Overview](#overview)
+    - [In-System Update of the Bootloader](#in-system-update-of-the-bootloader)
+    - [Programming the SMC with an External Programmer](#programming-the-smc-with-an-external-programmer)
+    
+- [I2C API](#i2c-api)
+    - [Command 0x80 = Transmit (master write)](#command-0x80--transmit-master-write)
+    - [Command 0x81 = Commit (master read)](#command-0x81--commit-master-read)
+    - [Command 0x82 = Reboot (master write)](#command-0x82--reboot-master-write)
+    - [Command 0x83 = Get bootloader version (master read)](#command-0x83--get-bootloader-version-master-read)
+    - [Command 0x84 = Rewind target address (master write)](#command-0x84--rewind-target-address-master-write)
+    - [Command 0x85 = Read flash memory](#command-0x85--read-flash-memory)
 
-The bootloader makes it possible to update the SMC firmware from the Commander X16 without an external programmer.
+# Project Description
 
+This is a custom bootloader for the Commander X16 ATtiny861 based System Management Controller (SMC).
 
-# How It Works
+The bootloader makes it possible to update the SMC firmware from the Commander X16 without using an external programmer.
 
-## Firmware Update Process
+The firmware is stored at the beginning of the flash memory and provides an interface to the PS/2 keyboard
+and mouse. It also handles the psysical push buttons and the LEDs of the system.
```

Comment:
This was the only typo I noticed while reading through (physical). (I try to be a little careful with spelling, after some "conflicts" at work when reviewing code/comments from dyslectic coworkers (vs my spelling OCD...)).